src/schoology-extractor/schoology_extractor/main.py

```diff
@@ -13,35 +13,66 @@
 load_dotenv()
 schoology_key = os.getenv("SCHOOLOGY_KEY")
 schoology_secret = os.getenv("SCHOOLOGY_SECRET")
-schoology_section_ids = os.getenv("SCHOOLOGY_SECTION_IDS")
 schoology_output_path = os.getenv("SCHOOLOGY_OUTPUT_PATH")
+schoology_grading_periods = os.getenv("SCHOOLOGY_GRADING_PERIODS")
+
+assert schoology_key is not None, "A `SCHOOLOGY_KEY` must be present in the .env file and it was not found."
+assert schoology_secret is not None, "A `SCHOOLOGY_SECRET` must be present in the .env file and it was not found."
+assert schoology_output_path is not None, "A `SCHOOLOGY_OUTPUT_PATH` must be present in the .env file and it was not found."
+assert schoology_grading_periods is not None, "A `SCHOOLOGY_GRADING_PERIODS` must be present in the .env file and it was not found."
 
-sections_id_array = schoology_section_ids.split(',')
-request_client = RequestClient(schoology_key, schoology_secret)
 
+grading_periods_array = schoology_grading_periods.split(',')
+
+request_client = RequestClient(schoology_key, schoology_secret)
 
 # export users
-export_data.to_csv(request_client.get_users(), f"{schoology_output_path}/users.csv")
+users_response = request_client.get_users()
+users_list = []
+while True:
+    users_list = users_list + users_response.current_page_items
+    if users_response.get_next_page() is None:
+        break
 
+export_data.to_csv(users_list, f"{schoology_output_path}/users.csv")
```

Comment:
Suggest replacing these with [os.path.join()](https://docs.python.org/3/library/os.path.html#os.path.join)

Reply:
It is probably a good idea. Personally I've been very lax about that, since PowerShell is perfectly happy to accept forward slashes. However, you're right that it _is_ a better practice: with the forward slash, we're _forcing_ people to use PowerShell instead of `cmd.exe`.